Microsoft.Toolkit.Uwp.UI.Controls/MarkdownTextBlock/MarkdownTextBlock.cs

```diff
@@ -21,32 +46,61 @@ public MarkdownTextBlock()
 
             // Register for property callbacks that are owned by our parent class.
-            RegisterPropertyChangedCallback(FontSizeProperty, OnPropertyChanged);
-            RegisterPropertyChangedCallback(BackgroundProperty, OnPropertyChanged);
-            RegisterPropertyChangedCallback(BorderBrushProperty, OnPropertyChanged);
-            RegisterPropertyChangedCallback(BorderThicknessProperty, OnPropertyChanged);
-            RegisterPropertyChangedCallback(CharacterSpacingProperty, OnPropertyChanged);
-            RegisterPropertyChangedCallback(FontFamilyProperty, OnPropertyChanged);
-            RegisterPropertyChangedCallback(FontSizeProperty, OnPropertyChanged);
-            RegisterPropertyChangedCallback(FontStretchProperty, OnPropertyChanged);
-            RegisterPropertyChangedCallback(FontStyleProperty, OnPropertyChanged);
-            RegisterPropertyChangedCallback(FontWeightProperty, OnPropertyChanged);
-            RegisterPropertyChangedCallback(ForegroundProperty, OnPropertyChanged);
-            RegisterPropertyChangedCallback(PaddingProperty, OnPropertyChanged);
-            RegisterPropertyChangedCallback(RequestedThemeProperty, OnPropertyChanged);
-            RegisterPropertyChangedCallback(SchemeListProperty, OnPropertyChanged);
+            FontSizePropertyToken = RegisterPropertyChangedCallback(FontSizeProperty, OnPropertyChanged);
+            BackgroundPropertyToken = RegisterPropertyChangedCallback(BackgroundProperty, OnPropertyChanged);
+            BorderBrushPropertyToken = RegisterPropertyChangedCallback(BorderBrushProperty, OnPropertyChanged);
+            BorderThicknessPropertyToken = RegisterPropertyChangedCallback(BorderThicknessProperty, OnPropertyChanged);
+            CharacterSpacingPropertyToken = RegisterPropertyChangedCallback(CharacterSpacingProperty, OnPropertyChanged);
+            FontFamilyPropertyToken = RegisterPropertyChangedCallback(FontFamilyProperty, OnPropertyChanged);
+            FontStretchPropertyToken = RegisterPropertyChangedCallback(FontStretchProperty, OnPropertyChanged);
+            FontStylePropertyToken = RegisterPropertyChangedCallback(FontStyleProperty, OnPropertyChanged);
+            FontWeightPropertyToken = RegisterPropertyChangedCallback(FontWeightProperty, OnPropertyChanged);
+            ForegroundPropertyToken = RegisterPropertyChangedCallback(ForegroundProperty, OnPropertyChanged);
+            PaddingPropertyToken = RegisterPropertyChangedCallback(PaddingProperty, OnPropertyChanged);
+            RequestedThemePropertyToken = RegisterPropertyChangedCallback(RequestedThemeProperty, OnPropertyChanged);
+
         }
 
-        private void ThemeListener_ThemeChanged(Helpers.ThemeListener sender)
+        private void OnUnloaded(object sender, RoutedEventArgs e)
         {
-            RenderMarkdown();
+            UnhookListeners();
+            themeListener.ThemeChanged -= ThemeListener_ThemeChanged;
+            themeListener = null;
```

Comment:
This is definitely heading in the right direction in that it won't leak the entire MarkDownTextblock anymore, (which is the biggest memory cost), but it will still leak ThemeListeners as even though they are nulled here - they are still subscribed to global application events and so are never garbage collected. 

I'd imagine ThemeListener should be made to implement IDisposable, and it's dispose method should unhook the three events it hooks on too in it's constructor.